blog/02-02-2023-on-chain-randomness-with-celo-using-subgraphs/on-chain-randomness-with-celo-using-subgraphs.md

```diff
@@ -0,0 +1,1867 @@
+    }else {
+        club.type = "NFT";
+        ClubNFT.create(event.params.club);
+    }
+    club.createdAtTimestamp = event.block.timestamp;
+    club.createdAtBlockNumber = event.block.number;
+    club.save();
+    clubList.push(club.id);
+    manager.club = clubList;
+    manager.save();
+}
+
+export function handleWinerToken(event: WinerToken):void {
+    let winerId = event.transaction.hash.toHexString() + "-" + event.logIndex.toString();
+    let winer = new Winers(winerId);
+    winer.address = event.params.winer_;
+    winer.timestamp = event.block.timestamp;
+    winer.blockNumber = event.block.number;
+
+    let club = Club.load(event.params.club.toHexString());
+    winer.club = club!.id;
+    winer.save();
+}
+
+export function handleWinerNFT(event: WinerNFT):void {
+    let winerId = event.transaction.hash.toHexString() + "-" + event.logIndex.toString();
+    let winer = new Winers(winerId);
+    winer.address = event.params.winer_;
+    winer.timestamp = event.block.timestamp;
+    winer.blockNumber = event.block.number;
+
+    let club = Club.load(event.params.club_.toHexString());
+    winer.club = club!.id;
+    winer.save();
+}
+
+export function handleRewardUpdateToken(event: RewardToken):void {
+    let club = Club.load(event.params.club.toHexString());
+    club!.reward = event.params.reward_;
+    club!.deposit = event.params.deposit_;
+    club!.membersLimit = event.params.membersLimit_;
+    club!.save();
+}
+
+export function handleNFTReward(event: RewardNFT):void {
+    let club = Club.load(event.params.club_.toHexString());
+    club!.reward = event.params.reward_;
+    club!.rewardAddress = event.params.rewardAddress_;
+    club!.save();
+}
+
+export function handleLimitDepoNFTUpdate(event: LimitAndDepoUpdate):void {
+    let club = Club.load(event.params.club_.toHexString());
+    if(event.params.name_ == "membersLimit") {
+        club!.membersLimit = event.params.update_;
+    }else {
+        club!.deposit = event.params.update_;
+    }
+    club!.save();
+}
+```
+
+### Build and Deploy
+
+Once we have completed the code generation process, our next step is to build and deploy the subgraphs. To accomplish this, we can execute the following command in the terminal.
+
+```bash
+# Build
+graph build
+
+
+# Deploy
+graph deploy --product hosted-service <GITHUB_USER>/<SUBGRAPH_NAME>
+```
+
+After deployment, we can access our subgraphs on the dashboard page and test our queries.
+
+![Subgraph Dashboard](images/17.png)
+
+
+## Conclusion
+
+This tutorial teaches how to use a `subgraph` for querying data on the Celo blockchain. Additionally, it covers how to generate random numbers using Celo's provided RNG called `Celo Randomness`.
+
+
+## Next Step
+
+Moving forward, we can not only create but also use existing subgraphs to facilitate our access to data on the Celo blockchain. Additionally, we can use Celo Randomness to create a game that utilizes random numbers, which helps us prevent cheating.
```

Comment:
Can you add links to other topic related celo sage tutorials here, to guide the reader to read more tutorials on the celo sage